src/game/engine/overlay.cpp

```diff
@@ -0,0 +1,196 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
```

Comment:
Can we have a proper brief here and in the header, something like "Class for handling placement of overlay." I think I've let this slide too much.